lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -831,25 +831,24 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.debug("make_mpiexec_hydra_options: no rmk from HYDRA_RMK %s and hydra_info %s",
                                self.HYDRA_RMK, self.hydra_info)
-        else:
-            launcher = None
-            if getattr(self, 'HYDRA_LAUNCHER', None) is not None:
-                launcher = [x for x in self.HYDRA_LAUNCHER if x in self.hydra_info.get('launcher', [])]
-                if launcher:
-                    self.log.debug("make_mpiexec_hydra_options: HYDRA: launcher %s, using first one", launcher)
-                else:
-                    self.log.debug("make_mpiexec_hydra_options: no launcher from HYDRA_LAUNCHER %s and hydra_info %s",
-                                   self.HYDRA_LAUNCHER, self.hydra_info)
-
-            launcher_exec = self.HYDRA_LAUNCHER_EXEC
-            if not launcher:
-                launcher_exec = self.get_rsh()
+        launcher = None
+        if getattr(self, 'HYDRA_LAUNCHER', None) is not None:
+            launcher = [x for x in self.HYDRA_LAUNCHER if x in self.hydra_info.get('launcher', [])]
```

Comment:
use `launcher` rather than `self.HYDRA_LAUNCHER`?

also, redefining `launcher` here makes this harder to understand...